set_goenv.sh

```diff
@@ -33,19 +32,11 @@ set_goenv() {
             ;;
     esac
 
-    # XXX Setting the compiler might not be necessary anymore now that we're
-    # using standard Go toolchain and if we don't put mongodbtoolchain into the
-    # path.  But if we need to keep mongodbtoolchain for other tools (eg. python),
-    # then this is probably still necessary to find the right gcc.
+    # On s390x, the Go toolchain relies on s390x-linux-gnu-gcc which isn't set up.
+    # Temporarily rely on the mongodbtoolchain for compilation on s390x.
     if [ -z "$CC" ]; then
         UNAME_M=$(PATH="/usr/bin:/bin" uname -m)
         case $UNAME_M in
-            aarch64)
-                export CC=/opt/mongodbtoolchain/v3/bin/aarch64-mongodb-linux-gcc
-            ;;
-            ppc64le)
-                export CC=/opt/mongodbtoolchain/v3/bin/ppc64le-mongodb-linux-gcc
-            ;;
             s390x)
                 export CC=/opt/mongodbtoolchain/v3/bin/s390x-mongodb-linux-gcc
```

Comment:
out of curiosity, does it fix the s390x issue if you just change this to `gcc`?

Reply:
I think so: https://evergreen.mongodb.com/version/601323753e8e86520af94e4a

Hard to technically say for all the versions since it'll take a very long time for them to run, but all s390x's should be the same in this regard. Better to use `/usr/bin/gcc` then?

Reply:
If we can get rid of all the toolchain uses in one fell swoop, might as well!

Reply:
Sounds good! I added a new commit below if everything looks good to you.